client/layout.js

```diff
@@ -78,15 +167,9 @@ $.fn.scrollToCenter = function(speed) {
   speed = speed || 200;
   var el = this;
   var elOffset = el.offset().top;
-  var elHeight = el.height();
   var windowHeight = $(window).height();
-  var offset;
+  var offset = elOffset - Math.max((windowHeight - el.height()) / 2, 0);
```

Comment:
Yeah, stackoverflow. I don't have an account, and trying to contribute to that website frustrates the hell out of me, so I won't be pushing it upstream.